taxonomy/utils.py

```diff
@@ -234,17 +234,24 @@ def process_skill_attr_text(text_data: str, product_type: ProductTypes) -> dict:
     return extra_data
 
 
-def skip_product_processing(extra_data: dict, key_or_uuid: str, product_type: ProductTypes) -> bool:
+def xblock_with_same_content(extra_data: dict, key_or_uuid: str):
     """
-    Check whether to skip processing.
+    Return identifier of the similar xblock if the content is same.
```

Comment:
@navinkarkera I was first wondering why you are using `first()` here instead of `all()` as, we would want to tag all the XBlocks with the same content. Then I realized, you are actually not fetching just any XBlock that has similar content, but actually the XBlock with tags already on them, hence `auto_processed=True` in the filter. The XBlocks fetched here are not the destinations, but the source of the tagging.

But the function or the docstring is not clear about this. Can you kindly specify more explicitly that this function fetches an already tagged XBlock with the same content?

Reply:
@tecoholic Good point. Updated.